### PR TITLE
Allow leading zeros in tryParseInt

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Fixes https://buganizer.corp.google.com/issues/141298281

A customer noticed that we don't correctly order their documents. Some of their documents are using keys of the form "00x", where x is a 9 digit number. Our number check only allows for ten digits, and leading zeros can break this limit. This PR relaxes the check and matches the backend semantics.

### CODE REVIEW

File: packages/database/test/path.test.ts [hunk 1]
```diff
@@ -117,5 +117,6 @@ describe('Path Tests', function() {
     expectGreater('/ab', '/a');
     expectGreater('/a/b', '/a');
     expectGreater('/a/b', '/a//');
+    expectGreater('/a/0971500000', '/a/00403311635');
```
Comment (jsdt) [inline packages/database/test/path.test.ts]:
```suggestion
    expectGreater('/a/0971500000', '/a/00403311635');
    expectGreater('/a/0971500000', '/a/971500000');
```

Does this handle that case correctly (when both parse to the same Int, we fall back to comparing the length of the strings)?

Comment (schmidt-sebastian) [reply packages/database/test/path.test.ts]: I think it should, but let's find out :)

File: packages/database/test/path.test.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ import { Path } from '../src/core/util/Path';
 
 describe('Path Tests', function() {
   const expectGreater = function(left, right) {
-    expect(Path.comparePaths(new Path(left), new Path(right))).to.equal(1);
-    expect(Path.comparePaths(new Path(right), new Path(left))).to.equal(-1);
+    expect(Path.comparePaths(new Path(left), new Path(right))).to.be.greaterThan(0)
```
Comment (mikelehen) [inline packages/database/test/path.test.ts]: missing semicolon?

Comment (schmidt-sebastian) [reply packages/database/test/path.test.ts]: Well, on GitHub it is missing. But I swear that I have this locally (via the Prettier commit that once again got stuck somewhere...)

**Comment by schmidt-sebastian** [comment]
CI is passing now after a re-run.

**Pull request merged**